File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/json/Type.java
package org.kevoree.modeling.api.json;

/**
 * Created by duke on 10/3/14.
 */
public enum Type {

    VALUE(0),
    LEFT_BRACE(1),
    RIGHT_BRACE(2),
    LEFT_BRACKET(3),
    RIGHT_BRACKET(4),
    COMMA(5),
    COLON(6),
    EOF(42);

    private final int _value;

    private Type(int p_value) {
        this._value = p_value;
    }

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/json/Type.ts

class Type {

  public static VALUE: Type = new Type(0);
  public static LEFT_BRACE: Type = new Type(1);
  public static RIGHT_BRACE: Type = new Type(2);
  public static LEFT_BRACKET: Type = new Type(3);
  public static RIGHT_BRACKET: Type = new Type(4);
  public static COMMA: Type = new Type(5);
  public static COLON: Type = new Type(6);
  public static EOF: Type = new Type(42);
  private _value: number = 0;

  constructor(p_value: number) {
    this._value = p_value;
  }

  public equals(other: AccessMode): boolean {
        return this == other;
    }

}

